vendor/bat-native-ads/src/bat/ads/internal/bundle/creative_ad_info.cc

```diff
@@ -13,4 +14,24 @@ CreativeAdInfo::CreativeAdInfo(const CreativeAdInfo& info) = default;
 
 CreativeAdInfo::~CreativeAdInfo() = default;
 
+bool CreativeAdInfo::operator==(const CreativeAdInfo& rhs) const {
+  return creative_instance_id == rhs.creative_instance_id &&
+         creative_set_id == rhs.creative_set_id &&
+         campaign_id == rhs.campaign_id &&
+         start_at_timestamp == rhs.start_at_timestamp &&
+         end_at_timestamp == rhs.end_at_timestamp &&
+         daily_cap == rhs.daily_cap && advertiser_id == rhs.advertiser_id &&
+         priority == rhs.priority && ptr == rhs.ptr &&
+         conversion == rhs.conversion && per_day == rhs.per_day &&
+         per_week == rhs.per_week && per_month == rhs.per_month &&
+         total_max == rhs.total_max &&
+         split_test_group == rhs.split_test_group && segment == rhs.segment &&
+         CompareAsSets(geo_targets, rhs.geo_targets) &&
```

Comment:
geo_targets data structure should have the == != overloads which internally use CompareAsSets